runelite-client/src/main/java/net/runelite/client/plugins/notes/NotesPanel.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.notes;
+
+import javax.swing.BorderFactory;
+import javax.swing.JEditorPane;
+import javax.swing.JLabel;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import javax.swing.text.BadLocationException;
+import javax.swing.text.Document;
+
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.client.ui.PluginPanel;
+
+import java.awt.BorderLayout;
+
+@Slf4j
+public class NotesPanel extends PluginPanel
+{
+	private final JEditorPane notesEditor = new JEditorPane();
+
+	void init(NotesConfig config)
+	{
+		// this may or may not qualify as a hack
+		// but this lets the editor pane expand to fill the whole parent panel
+		getParent().setLayout(new BorderLayout());
+		getParent().add(this, BorderLayout.CENTER);
+
+		setLayout(new BorderLayout());
+		setBorder(BorderFactory.createEmptyBorder(2, 6, 6, 6));
+
+		final JLabel notesHeader = new JLabel("Notes");
+		add(notesHeader, BorderLayout.NORTH);
+
+		notesEditor.setContentType("text/plain");
+
+		if (config.persistNotes())
+		{
+			// load note text
+			String data = unEscapeNote(config.notesData());
+			notesEditor.setText(data);
+
+			notesEditor.getDocument().addDocumentListener(new DocumentListener()
+			{
+				@Override
+				public void insertUpdate(DocumentEvent e)
+				{
+					notesChanged(e.getDocument());
+				}
+
+				@Override
+				public void removeUpdate(DocumentEvent e)
+				{
+					notesChanged(e.getDocument());
+				}
+
+				@Override
+				public void changedUpdate(DocumentEvent e)
+				{
+					notesChanged(e.getDocument());
+				}
+
+				private void notesChanged(Document doc)
+				{
+					try
+					{
+						// get document text and save to config whenever editor is changed
+						String data = escapeNote(doc.getText(0, doc.getLength()));
+						log.debug("Save Notes: " + data.length());
+						config.notesData(data);
+					}
+					catch (BadLocationException ex)
+					{
+						log.warn("Notes Document Bad Location: " + ex);
+					}
+				}
+			});
+		}
+		add(notesEditor, BorderLayout.CENTER);
+	}
+
+	private String escapeNote(String s)
+	{
+		return s.replaceAll("\n", "\\\n");
```

Comment:
In what was does it not work without escaping? afaik the properties file and config service both would work ok without it.

Reply:
You are right, I just looked at the format of settings.properties and assumed there shouldn't be line breaks in it. Look like you've already covered this.

Reply:
properties is part of java: https://docs.oracle.com/javase/tutorial/essential/environment/properties.html